### PR TITLE
Add Multi Translate Extension

### PR DESCRIPTION
**Pull request by djmbritt**
Hi, there this is the extension that I've been prototyping for the last couple of days.
It's working pretty well, and I'm pretty sure I've managed to catch most exceptions. 

So a couple of screen shots to see  how it looks like:
When the trigger is initiated:

image

The list of languages is dynamically updated based on the current language configuration.
The default list and order of languages is based on the number of speakers of each language.

This is how it looks like when a user types a query:
image
As you can see there are two actions namely; copy to clipboard and open in a browser.

As discussed we discussed on IRC on Tuesday I create a directory and configuration file, which is: "~/.config/albert/org.albert.extension.mtr/language_config.json"
The script creates this directory and file on the fly the first time it is initiated. 
Afterwords the user should be able to locate the file and add or remove languages as they please.
How it looks like after a user has changed the language_config.json file:
image
The only problem with this approach is that the user needs to relaunch Albert.
But it should be persistent afterwards.

I also added a modifier, so that user's can open up the configuration file via Albert.
image

The user is informed when there is a connection issue.
image
This is initiated when the user starts typing a query and an exception is raised.

### CODE REVIEW

File: MultiGoogleTranslate.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+__version__ = "1.0"
+__trigger__ = "mtr"
+__author__ = "David Britt, Manuel Schneider"
+__dependencies__ = []
+
+ua = "Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/62.0.3202.62 Safari/537.36"
+urltmpl = "https://translate.googleapis.com/translate_a/single?client=gtx&sl=auto&tl=%s&dt=t&q=%s"
+urlbrowser = "https://translate.google.com/#auto/%s/%s"
+
+configuration_directory = os.path.expanduser("~/.config/albert/org.albert.extension.mtr")
```
Comment (ManuelSchneid3r) [inline MultiGoogleTranslate.py]: please use [Extension interface (0.2)](https://albertlauncher.github.io/docs/extensions/python/#extension-interface-02) to get the config path.

File: MultiGoogleTranslate.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+import json
+import urllib.request
+import urllib.parse
+import os
+
+__iid__ = "PythonInterface/v0.1"
+__prettyname__ = "Multi Google Translate"
+__version__ = "1.0"
+__trigger__ = "mtr"
+__author__ = "David Britt, Manuel Schneider"
```
Comment (ManuelSchneid3r) [inline MultiGoogleTranslate.py]: hmm, this is your work :D

File: MultiGoogleTranslate.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+Add or remove elements based on the ISO-Codes that you found on the google documentation page.
+"""
+
+from albertv0 import *
+import json
+import urllib.request
+import urllib.parse
+import os
+
+__iid__ = "PythonInterface/v0.1"
```
Comment (ManuelSchneid3r) [inline MultiGoogleTranslate.py]: needs to be 0.2 for the changes in L.29

File: MultiGoogleTranslate.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+__prettyname__ = "Multi Google Translate"
+__version__ = "1.0"
+__trigger__ = "mtr"
```
Comment (ManuelSchneid3r) [inline MultiGoogleTranslate.py]: please append a space to the extensions trigger to avoid collisions. the extension consumes arguments anyway

**Comment by djmbritt** [comment]
Thaks for the feedback. 
Will get on top of it.

**Comment by djmbritt** [comment]
I've changed what was asked and I've also added the folder, file creation in the initialize() function.

**Comment by djmbritt** [comment]
I'm guessing I can close this now?

**Comment by ManuelSchneid3r** [comment]
> I'm guessing I can close this now?

The PR will be closed automatically as soon it is merged.

> I've changed what was asked and I've also added the folder, file creation in the initialize() function.

I have often have the problem that I get no notification when requested changes in a PR have been implemented. The problem is that I dont know how Github calls this. Can you please check it and send a screenshot how it looks like? IIRC it is possible.

File: dangokao/__init__.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+		if len(query.string) >= 2:
+			try:
+				with requests.get(dangoUrl, params={"q": query.string}) as api_response:
+					json_data = json.loads(api_response.text)
+					for emoj in json_data["items"]:
+						results.append(
+							Item(
+								    id=__prettyname__,
+                                    icon=icon_path,
```
Comment (ManuelSchneid3r) [inline dangokao/__init__.py]: Does python accept this kind of identation?

Comment (djmbritt) [reply dangokao/__init__.py]: That's not right...
Pretty sure it was working properly when I using the script.
But now I'm not so sure.

**Comment by djmbritt** [comment]
So i checked the indentation, it was fine.
But I passed it through an autoformatter just in case.

File: dangoemoji/__init__.py [hunk 6]
```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8 -*-
+
+"""
+Use machinelearning to find the relevant emoji.
```
Comment (ManuelSchneid3r) [inline dangoemoji/__init__.py]: ml? where? :smile:

**Comment by ManuelSchneid3r** [comment]
Next time please try to separate PR. Thank you @djmbritt these extensions seem to be pretty valuable. :+1:

**Comment by ManuelSchneid3r** [comment]
Get errors though:
```
17:50:09 [WARN] [dangoemoji] Error: [('SSL routines', 'SSL_CTX_new', 'library has no ciphers')]

At:
  /usr/lib/python3.6/site-packages/OpenSSL/_util.py(48): exception_from_error_queue
  /usr/lib/python3.6/site-packages/OpenSSL/_util.py(61): openssl_assert
  /usr/lib/python3.6/site-packages/OpenSSL/SSL.py(474): __init__
  /usr/lib/python3.6/site-packages/urllib3/contrib/pyopenssl.py(379): __init__
  /usr/lib/python3.6/site-packages/urllib3/util/ssl_.py(254): create_urllib3_context
  /usr/lib/python3.6/site-packages/urllib3/connection.py(314): connect
  /usr/lib/python3.6/site-packages/urllib3/connectionpool.py(850): _validate_conn
  /usr/lib/python3.6/site-packages/urllib3/connectionpool.py(350): _make_request
  /usr/lib/python3.6/site-packages/urllib3/connectionpool.py(658): urlopen
  /usr/lib/python3.6/site-packages/requests/adapters.py(523): send
  /usr/lib/python3.6/site-packages/requests/sessions.py(618): send
  /usr/lib/python3.6/site-packages/requests/sessions.py(508): request
  /usr/lib/python3.6/site-packages/requests/api.py(58): request
  /usr/lib/python3.6/site-packages/requests/api.py(72): get
  /home/manuel/.local/share/albert/org.albert.extension.python/modules/dangoemoji/__init__.py(80): handleQuery
.
```

**Comment by marek-mazur** [comment]
It doesn't work...
```
13:59:57 [WARN] [MultiGoogleTranslate] TypeError: the JSON object must be str, not 'bytes'

At:
  /usr/lib/python3.5/json/__init__.py(312): loads
  /usr/lib/python3.5/json/__init__.py(268): load
  /usr/share/albert/org.albert.extension.python/modules/MultiGoogleTranslate.py(93): handleQuery
```

**Comment by ManuelSchneid3r** [comment]
Wad? Min python should be 3.5 :D Works on my machine with Python 3.6.4

**Comment by marek-mazur** [comment]
I have exactly the same problem on all my machines with your plugin [GoogleTranslate.py](https://github.com/albertlauncher/python/blob/master/GoogleTranslate.py)
The solution is to do:
```
import codecs

reader = codecs.getreader("utf-8")
data = json.load(reader(response))
```

instead of:

```
data = json.load(response)
```

**Pull request merged**